src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -75,7 +75,7 @@ internal virtual bool OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerO
         }
 
         // Provide a default implementation for value converters.
-        internal virtual bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, ref ReadStack state, out T value)
+        internal virtual bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, ref ReadStack state, [MaybeNullWhen(false)] out T value)
```

Comment:
I don't know if this is accurate. Why was this annotation necessary?
- `T` could be null when true (imagine the payload contains `{"foo": null}`)
- `T` could be non-null when returning true
- `T` could be null when false

**Questions:**
- I don't know whether `T` can be non-null when returning false or if it is guaranteed to be null when returning false
- I don't know when `OnTryRead` is intended to return false.

@steveharter - what is the intent here of this design and how do components that override this virtual behave?

Reply:
I added this because all implementations of this virtual except: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/KeyValuePairConverter.cs#L27 have this attribute and roslyn was complaining on the ones that had it because it didn't math the overriden member.

To me this makes the most sense since it is a virtual. This is saying, even if the `T` is non-nullable it might be null when returning false. 

cc: @stephentoub

Reply:
> To me this makes the most sense since it is a virtual. This is saying, even if the `T` is non-nullable it might be null when returning false. 

Sounds good to me. Thanks.